kernel/memory_structs/src/lib.rs

```diff
@@ -24,6 +26,53 @@ use paste::paste;
 use derive_more::*;
 use range_inclusive::{RangeInclusive, RangeInclusiveIterator};
 
+/// Numeric constants associated with different page types.
+pub const PAGE_4KB_SIZE: usize = 1 << 12;
+pub const PAGE_2MB_SIZE: usize = (1 << 12) * 512;
+pub const PAGE_1GB_SIZE: usize = (1 << 12) * 512 * 512;
```

Comment:
Doc comments only apply to the single item immediately following them, so you need to do this:
```suggestion
/// The size in bytes of a normal 4KiB page: 4096 bytes.
pub const PAGE_4KB_SIZE: usize = 1 << 12;
/// The size in bytes of a huge 2MiB page: 2097152 bytes.
pub const PAGE_2MB_SIZE: usize = (1 << 12) * 512;
/// The size in bytes of a normal 1GiB page: 1073741824 bytes.
pub const PAGE_1GB_SIZE: usize = (1 << 12) * 512 * 512;
```

Can you also use the definitions from the `kernel_config::memory` module:
* There is a const called `PAGE_SIZE` which is the same as your `PAGE_4KB_SIZE`
  * we can and will rename that `PAGE_SIZE` definition later, just in a separate PR
* There is a const called `ENTRIES_PER_PAGE_TABLE`, which has the value of 512.

Take a look at that module and see if any other consts better represent the semantics behind how you calculate the page size consts here.